From 5ab6fb580b62ddcccad29f05fcc678fa14af4707 Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Thu, 28 Jan 2021 18:13:28 +0100
Subject: [PATCH 143/281] atm: eni: Use tasklet_disable_in_atomic() in the
 ->send() callback

The atmdev_ops::send callback may be invoked with disabled BH from
net_device_ops::ndo_start_xmit().

Use tasklet_disable_in_atomic() here so it can continue spin in atomic
context. The other user of tasklet_disable() are preemptible so they can
remain.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 drivers/atm/eni.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/atm/eni.c b/drivers/atm/eni.c
index 316a994..e96a4e8 100644
--- a/drivers/atm/eni.c
+++ b/drivers/atm/eni.c
@@ -2054,7 +2054,7 @@ static int eni_send(struct atm_vcc *vcc,struct sk_buff *skb)
 	}
 	submitted++;
 	ATM_SKB(skb)->vcc = vcc;
-	tasklet_disable(&ENI_DEV(vcc->dev)->task);
+	tasklet_disable_in_atomic(&ENI_DEV(vcc->dev)->task);
 	res = do_tx(skb);
 	tasklet_enable(&ENI_DEV(vcc->dev)->task);
 	if (res == enq_ok) return 0;
-- 
2.7.4

